test/integration/authrep/application_keys_test.rb

```diff
@@ -12,80 +12,126 @@ def setup
+    @application_oauth = Application.save(:service_id => @service_oauth.id,
+                                          :id         => next_id,
+                                          :state      => :active,
+                                          :plan_id    => @plan_id,
+                                          :plan_name  => @plan_name)
   end
 
   test_authrep 'succeeds if no application key is defined nor passed' do |e|
     get e, :provider_key => @provider_key,
-           :app_id       => @application.id
+           :service_id   => @service_oauth.id,
+           :app_id       => @application_oauth.id
 
     assert_authorized
   end
 
   test_authrep 'succeeds if one application key is defined and the same one is passed' do |e|
-    application_key = @application.create_key
+    application_key = @application_oauth.create_key
 
     get e, :provider_key => @provider_key,
-           :app_id       => @application.id,
+           :service_id   => @service_oauth.id,
+           :app_id       => @application_oauth.id,
            :app_key      => application_key
 
     assert_authorized
   end
 
   test_authrep 'succeeds if multiple application keys are defined and one of them is passed' do |e|
-    application_key_one = @application.create_key
-    _application_key_two = @application.create_key
+    application_key_one = @application_oauth.create_key
+    _application_key_two = @application_oauth.create_key
 
     get e, :provider_key => @provider_key,
-           :app_id       => @application.id,
+           :service_id   => @service_oauth.id,
+           :app_id       => @application_oauth.id,
            :app_key      => application_key_one
 
     assert_authorized
   end
 
-  test_authrep 'does not authorize if application key is defined but not passed',
+  test_authrep 'does not authorize if application key is defined but not passed for non oauth services',
                except: :oauth_authrep do |e|
     @application.create_key
 
     get e, :provider_key => @provider_key,
+           :service_id   => @service.id,
            :app_id       => @application.id
 
     assert_not_authorized 'application key is missing'
   end
 
+  test_authrep 'authorizes if application key is defined but not passed if the service is oauth' do |e|
+    @application_oauth.create_key
+
+    get e, :provider_key => @provider_key,
+           :service_id   => @service_oauth.id,
+           :app_id       => @application_oauth.id
+
+    assert_authorized
+  end
+
+  test_authrep 'authorizes if application key is defined and matches if the service is oauth' do |e|
+    application_key = @application_oauth.create_key
+
+    get e, :provider_key => @provider_key,
+           :service_id   => @service_oauth.id,
+           :app_id       => @application_oauth.id,
+           :app_key      => application_key
+
+    assert_authorized
+  end
+
+  test_authrep 'does not authorize if application key is defined and passed even if the service is oauth',
```

Comment:
I think this description needs to mention that the key passed is invalid.

Reply:
Also, I don't understand the `except: :oauth_authrep` part :thinking:

Reply:
oauth_authrep only works with oauth services (ok) and never checks app_keys (nope, the check is about an invalid app_key but it will just authorize if calling oauth_authrep), so it's pointless to test oauth_authrep.

Reply:
re: the description, yes, should mention it is invalid.